SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -1113,6 +1147,158 @@ FString FSpatialGDKEditorToolbarModule::GetOptionalExposedRuntimeIP() const
+			EAppReturnType::Type UserAnswer = FMessageDialog::Open(EAppMsgType::YesNo, FText::FromString(MissingSimPlayerBuildText));
+			if (UserAnswer == EAppReturnType::No || UserAnswer == EAppReturnType::Cancel)
+			{
+				OnShowFailedNotification("Failed to launch cloud deployment. path not exist");
+				return;
+			}
+		}
+	}
+
+	auto LaunchCloudDeployment = [this]()
+	{
+		SpatialGDKEditorInstance->LaunchCloudDeployment(
+			FSimpleDelegate::CreateLambda([this]()
+			{
+				this->OnShowSuccessNotification("Successfully launched cloud deployment.");
+			}),
+
+			FSimpleDelegate::CreateLambda([this]()
+			{
+				this->OnShowFailedNotification("Failed to launch cloud deployment. See output logs for details.");
+			})
+		);
+	};
+
+#if ENGINE_MINOR_VERSION <= 22
+	AttemptSpatialAuthResult = Async<bool>(EAsyncExecution::Thread, []() { return SpatialCommandUtils::AttemptSpatialAuth(GetDefault<USpatialGDKSettings>()->IsRunningInChina()); },
+#else
+	AttemptSpatialAuthResult = Async(EAsyncExecution::Thread, []() { return SpatialCommandUtils::AttemptSpatialAuth(GetDefault<USpatialGDKSettings>()->IsRunningInChina()); },
+#endif
+		[this, LaunchCloudDeployment]()
+	{
+		if (AttemptSpatialAuthResult.IsReady() && AttemptSpatialAuthResult.Get() == true)
+		{
+			LaunchCloudDeployment();
+		}
+		else
+		{
+			OnShowTaskStartNotification(TEXT("Spatial auth failed attempting to launch cloud deployment."));
+		}
+	});
+}
+
+bool FSpatialGDKEditorToolbarModule::IsDeploymentConfigurationValid() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
+	return SpatialGDKSettings->GetPrimaryDeploymentName().IsEmpty() || SpatialGDKSettings->GetAssemblyName().IsEmpty() ? false : true;
+}
+
+bool FSpatialGDKEditorToolbarModule::CanBuildAndUpload() const
+{
+	return SpatialGDKEditorInstance->GetPackageAssemblyRef()->CanBuild();
+}
+
+bool FSpatialGDKEditorToolbarModule::CanLaunchDeployment() const
+{
+	return IsDeploymentConfigurationValid() && CanBuildAndUpload();
+}
+
+bool FSpatialGDKEditorToolbarModule::IsSimulatedPlayersEnabled() const
+{
+	return GetDefault<USpatialGDKEditorSettings>()->IsSimulatedPlayersEnabled();
+}
+
+void FSpatialGDKEditorToolbarModule::OnCheckedSimulatedPlayers()
+{
+	GetMutableDefault<USpatialGDKEditorSettings>()->SetSimulatedPlayersEnabledState(!IsSimulatedPlayersEnabled());
+}
+
+bool FSpatialGDKEditorToolbarModule::IsBuildClientWorkerEnabled() const
+{
+	return GetDefault<USpatialGDKEditorSettings>()->IsBuildClientWorkerEnabled();
+}
+
+void FSpatialGDKEditorToolbarModule::OnCheckedBuildClientWorker()
+{
+	GetMutableDefault<USpatialGDKEditorSettings>()->SetBuildClientWorker(!IsBuildClientWorkerEnabled());
+}
+
+void FSpatialGDKEditorToolbarModule::AddDeploymentTag(const FString& Tag)
```

Comment:
The naming doesn't make it clear that we check if the flag is present and add it otherwise.
```suggestion
void FSpatialGDKEditorToolbarModule::AddDeploymentTagIfNotPresent(const FString& Tag)
```